### PR TITLE
Add `MapDatasetEventSampler` event_list_meta() and run() method

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR aims at adding the methods `run()` and `event_list_meta()` to the `~gammapy.cube.MapDatasetEventSampler` class. 
The `run` method samples events from the background and sources, and applies IRF corrections to the latters. The Edisp can be switched off if necessary. The output is a `~gammapy.data.EventList` object.
Metadata information are stored through the `event_list_meta()` method.

### CODE REVIEW

**Comment by adonath** [comment]
Thanks @fabiopintore! I was curious and just tried the `MapDatasetEventSampler` locally. After I fixed a little bug in `EDispMap.sample_coord()` (attached to this PR so please fetch an rebase...) it seemed to work nicely. I'll proceed now with the review...

File: gammapy/cube/simulate.py [hunk 1]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+        meta["EV_CLASS"] = ""
+        meta["TELAPSE"] = ""
+
+        meta["GEOLON"] = ""
+        meta["GEOLAT"] = ""
+        meta["ALTITUDE"] = ""
+
+        return meta
+
+    def run(self, dataset, observation=None, edisp="True"):
```
Comment (adonath) [inline gammapy/cube/simulate.py]: Please remove the `edisp` parameter here. I think always adding the `ENERGY_TRUE` column is a good default for now. We will use it a lot for debugging and testing. Once the `MapDatasetEventSampler` is production ready, we can make it an option.

File: gammapy/cube/simulate.py [hunk 2]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+
+        Returns
+        -------
+        meta : dict
+            Meta dictionary.
+        """
+        # See: https://gamma-astro-data-formats.readthedocs.io/en/latest/events/events.html#mandatory-header-keywords
+        meta = {}
+        # TODO: currently additional meta data is missing
+        import gammapy
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py [hunk 3]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+        """
+        events_bkg = self.sample_background(dataset)
+        events_src = self.sample_sources(dataset)
+        events_src = self.sample_psf(dataset.psf, events_src)
+
+        if edisp == "True":
+            events_src = self.sample_edisp(dataset.edisp, events_src)
+
+        events = EventList.stack([events_bkg, events_src])
+        #         events.table["EVENT_ID"] =
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py [hunk 4]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+        )
+        meta["TSTOP"] = (
+            ((dataset.gti.time_stop.mjd - dataset.gti.time_ref.mjd) * u.day)
+            .to(u.s)
+            .value[0]
+        )
+        meta["ONTIME"] = dataset.gti.time_sum.value
+        meta["LIVETIME"] = observation.gti.time_sum.value
+        meta["DEADC"] = dataset.gti.time_sum.value / observation.gti.time_sum.value
+        meta["RA_PNT"] = observation.pointing_radec.icrs.ra.value
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py [hunk 5]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+            .value[0]
+        )
+        meta["TSTOP"] = (
+            ((dataset.gti.time_stop.mjd - dataset.gti.time_ref.mjd) * u.day)
+            .to(u.s)
+            .value[0]
+        )
+        meta["ONTIME"] = dataset.gti.time_sum.value
+        meta["LIVETIME"] = observation.gti.time_sum.value
+        meta["DEADC"] = dataset.gti.time_sum.value / observation.gti.time_sum.value
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py [hunk 6]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+            .to(u.s)
+            .value[0]
+        )
+        meta["TSTOP"] = (
+            ((dataset.gti.time_stop.mjd - dataset.gti.time_ref.mjd) * u.day)
+            .to(u.s)
+            .value[0]
+        )
+        meta["ONTIME"] = dataset.gti.time_sum.value
+        meta["LIVETIME"] = observation.gti.time_sum.value
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py [hunk 7]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+            ((dataset.gti.time_start.mjd - dataset.gti.time_ref.mjd) * u.day)
+            .to(u.s)
+            .value[0]
+        )
+        meta["TSTOP"] = (
+            ((dataset.gti.time_stop.mjd - dataset.gti.time_ref.mjd) * u.day)
+            .to(u.s)
+            .value[0]
+        )
+        meta["ONTIME"] = dataset.gti.time_sum.value
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py [hunk 8]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+        meta["OBSERVER"] = ""
+        meta["CREATED"] = ""
+        meta["OBJECT"] = dataset.name
+        meta["RA_OBJ"] = ""
+        meta["DEC_OBJ"] = ""
+        meta["OBS_MODE"] = ""
+        meta["EV_CLASS"] = ""
+        meta["TELAPSE"] = ""
+
+        meta["GEOLON"] = ""
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py [hunk 9]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+        meta["TIMESYS"] = "TT"
+        meta["TIMEREF"] = "LOCAL"
+        #         meta["DATE-OBS"] = dataset.gti.time_start.isot[0][0:10]
+        #         meta["DATE-END"] = dataset.gti.time_stop.isot[0][0:10]
+        meta["TIME-OBS"] = dataset.gti.time_start.isot[0]
+        meta["TIME-END"] = dataset.gti.time_stop.isot[0]
+
+        meta["OBS_ID"] = observation.obs_id
+
+        meta["TSTART"] = (
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py [hunk 10]
```diff
@@ -233,3 +235,107 @@ def sample_psf(self, psf_map, events):
...
+        meta["ORIGIN"] = ""
+        meta["TELESCOP"] = ""
+        meta["INSTRUME"] = ""
+        #
+
+        meta["CREATOR"] = "Gammapy {}".format(gammapy.__version__)
+
+        meta["OBSERVER"] = ""
+        meta["CREATED"] = ""
+        meta["OBJECT"] = dataset.name
```
[the remark this answers is not in the document]
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

File: gammapy/cube/simulate.py (hunk 1 above)
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done.

**Pull request merged**